File: lab_2/sequence_generator.cpp
#include <iostream>
#include <vector>
#include <random>


std::vector<bool> SequenceGenerate(const size_t& size)
{
	std::random_device seed;
	std::mt19937 engine(seed());
	std::vector<bool> sequence(size);
	for (size_t i = 0; i < size; ++i)
		sequence[i] = engine() % 2;

	return sequence;
}


int main()
{
	std::vector<bool> binary_sequence;
	binary_sequence = SequenceGenerate(128);
	std::cout<< "Ѕинарна€ последовальность из 128 случайных битов: "
	for (auto bit : binary_sequence)
		std::cout << bit;
}

File: lab_2/SequenceGenerator.java
import java.util.Random;
import java.util.ArrayList;


 class SequenceGenerator {
    public static ArrayList<Boolean> GenerateSequence(int size) {
        Random engine = new Random();
        ArrayList<Boolean> sequence = new ArrayList<Boolean>();
        for (int i = 0; i < size; i++){
                    sequence.add(engine.nextBoolean());
        }

        return sequence;
    }


    public static void main(String[] args) {
        ArrayList<Boolean> binary_sequence = GenerateSequence(128);
        System.out.print("A sequence of random 128 bits: ")
        for (boolean bit : binary_sequence) {
            System.out.print(bit ? '1' : '0');
        }
    }
}